internal/os/disk/api.go

```diff
@@ -26,21 +24,63 @@ const (
 	IOCTL_STORAGE_QUERY_PROPERTY    = 0x002d1400
 )
 
-// Implements the OS API calls related to Disk Devices. All code here should be very simple
+// API declares the interface exposed by the internal API
+type API interface {
+	// ListDiskLocations - constructs a map with the disk number as the key and the DiskLocation structure
+	// as the value. The DiskLocation struct has various fields like the Adapter, Bus, Target and LUNID.
+	ListDiskLocations() (map[uint32]shared.DiskLocation, error)
+	// IsDiskInitialized returns true if the disk identified by `diskNumber` is initialized.
+	IsDiskInitialized(diskNumber uint32) (bool, error)
+	// InitializeDisk initializes the disk `diskNumber`
+	InitializeDisk(diskNumber uint32) error
+	// PartitionsExist checks if the disk `diskNumber` has any partitions.
+	PartitionsExist(diskNumber uint32) (bool, error)
+	// CreatePartitoin creates a partition in disk `diskNumber`
+	CreatePartition(diskNumber uint32) error
+	// Rescan updates the host storage cache (re-enumerates disk, partition and volume objects)
+	Rescan() error
+	// GetDiskNumberByName gets a disk number by `diskName`.
```

Comment:
diskName is page83 id? If so, put it in the comment..

Reply:
This API comes from v1alpha1 and it's no longer used from v1beta1, I checked that we're not using it in PD CSI, I don't know if diskName is page83

Reply:
Changed to:
```
	// GetDiskNumberByName gets a disk number by page83 ID (disk name)
	GetDiskNumberByName(page83ID string) (uint32, error)
```